src/rdkafka_admin.c

```diff
@@ -6666,3 +6773,705 @@ rd_kafka_DescribeConsumerGroups_result_groups(
+                    rd_list_cnt(cluster_authorized_operations), rd_free);
+                for (i = 0; i < rd_list_cnt(cluster_authorized_operations);
+                     i++) {
+                        int *entry =
+                            rd_list_elem(cluster_authorized_operations, i);
+                        int *oper = malloc(sizeof(int));
+                        *oper     = *entry;
+                        rd_list_add(clusterdesc->cluster_authorized_operations,
+                                    oper);
+                }
+        } else
+                clusterdesc->cluster_authorized_operations = NULL;
+
+        clusterdesc->node_cnt = md->broker_cnt;
+        clusterdesc->Nodes = rd_calloc(sizeof(rd_kafka_Node_t), md->broker_cnt);
+        for (i = 0; i < md->broker_cnt; i++) {
+                clusterdesc->Nodes[i].host = rd_strdup(md->brokers[i].host);
+                clusterdesc->Nodes[i].port = md->brokers[i].port;
+                clusterdesc->Nodes[i].id   = md->brokers[i].id;
+        }
+        return clusterdesc;
+}
+
+static void rd_kafka_ClusterDescription_destroy(
+    rd_kafka_ClusterDescription_t *clusterdesc) {
+        int i;
+        RD_IF_FREE(clusterdesc->cluster_id, rd_free);
+        if (clusterdesc->cluster_authorized_operations)
+                rd_list_destroy(clusterdesc->cluster_authorized_operations);
+        if (clusterdesc->Nodes) {
+                for (i = 0; i < clusterdesc->node_cnt; i++) {
+                        rd_kafka_Node_t *node = &(clusterdesc->Nodes[i]);
+                        RD_IF_FREE(node->host, rd_free);
+                        RD_IF_FREE(node->rack_id, rd_free);
+                }
+                RD_IF_FREE(clusterdesc->Nodes, rd_free);
+        }
+        rd_free(clusterdesc);
+}
+
+static void rd_kafka_ClusterDescription_free(void *ptr) {
+        rd_kafka_ClusterDescription_destroy(ptr);
+}
+/**
+ * @brief Send DescribeClusterRequest. Admin worker compatible callback.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_admin_DescribeClusterRequest(rd_kafka_broker_t *rkb,
+                                      const rd_list_t *topics /*(char*)*/,
+                                      rd_kafka_AdminOptions_t *options,
+                                      char *errstr,
+                                      size_t errstr_size,
+                                      rd_kafka_replyq_t replyq,
+                                      rd_kafka_resp_cb_t *resp_cb,
+                                      void *opaque) {
+        rd_kafka_resp_err_t err;
+        int include_cluster_authorized_operations;
+
+        include_cluster_authorized_operations =
+            rd_kafka_confval_get_int(&options->include_authorized_operations);
+
+        err = rd_kafka_MetadataRequest(
+            rkb, NULL, "describe cluster", rd_false /*no auto create*/,
+            include_cluster_authorized_operations,
+            rd_false /*!include topic authorized operations */,
```

Comment:
```suggestion
            rd_false /* don't include topic authorized operations */,
```

Reply:
Discuss with team to decide on a convention.

Reply:
After discussion: behaviour